recipe/pipeline.py

```diff
@@ -30,87 +30,47 @@ def source_url(day: str) -> str:
-            # for more. We'll have one output URL per day.
-            sources = source_url.map(self.days)
-
-            # Map the `download` task (provided by prefect) to download the raw data
-            # into a cache.
-            # Mapped outputs (sources) can be fed straight into another Task.map call.
-            # If an input is just a regular argument that's not a mapping, it must
-            # be wrapepd in `prefect.unmapped`.
-            # https://docs.prefect.io/core/concepts/mapping.html#unmapped-inputs
-            # nc_sources will be a list of cached URLs, one per input day.
-            nc_sources = download.map(sources, cache_location=unmapped(self.cache_location))
-
-            # The individual files would be a bit too small for analysis. We'll use
-            # pangeo_forge.utils.chunk to batch them up. We can pass mapped outputs
-            # like nc_sources directly to `chunk`.
-            chunked = pangeo_forge.utils.chunk(nc_sources, size=5)
-
-            # Combine all the chunked inputs and write them to their final destination.
-            writes = combine_and_write.map(
-                chunked,
-                unmapped(self.target_location),
-                append_dim=unmapped("time"),
-                concat_dim=unmapped("time"),
-            )
-
-            # Consolidate the metadata for the final dataset.
-            consolidate_metadata(self.target_location, writes=writes)
-
-        return flow
-
-
-# pangeo-forge and Prefect require that a `flow` be present at the top-level
-# of this module.
-flow = Pipeline().flow
+days = pd.date_range("1981-09-01", "1981-09-10", freq="D").strftime("%Y-%m-%d").tolist()
+sources = list(map(source_url, days))
+
+# Create the pangeo-forge recipe
+recipe = NetCDFtoZarrSequentialRecipe(
+    input_urls=sources,
+    sequence_dim="time"
+)
+this_dir = os.path.dirname(os.path.abspath(__file__))
+
+if args.storage == 's3':
+    # Read the config file
+    with open(f"{this_dir}/config.yml") as config_file:
+        config = yaml.safe_load(config_file)
+    fs = s3fs.S3FileSystem(key=config['MY_AWS_KEY'], secret=config['MY_AWS_SECRET'])
+    target_path = f"{config['s3']['target_bucket']}/{config['s3']['target_path']}/noaa_sst.zarr"
+    cache_path = f"{config['s3']['cache_bucket']}/{config['s3']['cache_path']}"
+else:
+    fs = LocalFileSystem()
+    target_path = os.path.join(this_dir, 'noaa_sst.zarr')
+    if os.path.exists(target_path):
+        shutil.rmtree(target_path)
+    os.mkdir(target_path)
+    cache_dir = tempfile.TemporaryDirectory()
+    cache_path = cache_dir.name
+
+target = FSSpecTarget(fs, root_path=target_path)
+cache_target = CacheFSSpecTarget(fs=fs, root_path=cache_path)
+recipe.input_cache = cache_target
+recipe.target = target
+pipeline = recipe.to_pipelines()
+
+if args.execution_env == 'prefect':
+    executor = PrefectPipelineExecutor()
+    print(executor)
+    plan = executor.pipelines_to_plan(pipeline)
+    # The 'plan' is a prefect.Flow
```

Comment:
Needed?

Reply:
IMO it's helpful to know that while the `plan` is a return from a pangeo_forge module, the return object is a prefect module (and so you can operate on it with prefect functions).

Reply:
Wont most editors/IDEs give you those prompts though?